src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransfer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import lombok.Data;
+import org.opensearch.knn.index.codec.util.SerializationMode;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ * Abstract class to transfer vector value from Java to native memory
+ */
+@Data
+public abstract class VectorTransfer {
+    protected final long vectorsStreamingMemoryLimit;
+    protected long totalLiveDocs;
+    protected long vectorsPerTransfer;
+    protected long vectorAddress;
+    protected int dimension;
+
+    public VectorTransfer(final long vectorsStreamingMemoryLimit) {
+        this.vectorsStreamingMemoryLimit = vectorsStreamingMemoryLimit;
+        this.vectorsPerTransfer = Integer.MIN_VALUE;
+    }
+
+    /**
+     * Initialize the transfer
+     *
+     * @param totalLiveDocs total number of vectors to be transferred
+     */
+    abstract public void init(final long totalLiveDocs);
+
+    /**
+     * Transfer a single vector
```

Comment:
the documentation is incorrect, we are not transferring 1 vector here. We are transferring more than 1 vector.

Reply:
I meant to say this method is called for a single vector.
Yes. We transfer more than 1 vector internally but it is hidden from caller.